lib/src/tar/export.rs

```diff
@@ -455,4 +466,47 @@ mod tests {
             assert!(symlink_is_denormal(path));
         }
     }
+
+    #[test]
+    fn test_write_repo_structure() {
+        use tar::EntryType::{Directory, Regular};
+
+        let mut expected = btreemap!(
+            "sysroot" => Directory,
+            "sysroot/ostree" => Directory,
+            "sysroot/ostree/repo" => Directory,
+            "sysroot/ostree/repo/refs" => Directory,
+            "sysroot/ostree/repo/objects/00" => Directory,
+            "sysroot/ostree/repo/objects/23" => Directory,
+            "sysroot/ostree/repo/objects/77" => Directory,
+            "sysroot/ostree/repo/objects/bc" => Directory,
+            "sysroot/ostree/repo/objects/ff" => Directory,
+            "sysroot/ostree/repo/tmp" => Directory,
+            "sysroot/ostree/repo/tmp/cache" => Directory,
+            "sysroot/ostree/repo/refs" => Directory,
+            "sysroot/ostree/repo/refs/heads" => Directory,
+            "sysroot/ostree/repo/refs/mirrors" => Directory,
+            "sysroot/ostree/repo/refs/remotes" => Directory,
+            "sysroot/ostree/repo/xattrs" => Directory,
+            "sysroot/ostree/repo/config" => Regular,
+        );
+
+        let mut archive = {
+            let mut builder = tar::Builder::new(vec![]);
+            let mut writer = OstreeTarWriter::new(&mut builder, ExportOptions::default());
+            writer.write_repo_structure().unwrap();
+            assert!(writer.wrote_initdirs);
+            let buf = builder.into_inner().unwrap();
+            tar::Archive::new(std::io::Cursor::new(buf))
+        };
+
+        for entry in archive.entries().unwrap() {
```

Comment:
There's a similar test here https://github.com/ostreedev/ostree-rs-ext/blob/392d5ed2d46c0fe7f966efc7705730356e620fa4/lib/tests/it/main.rs#L286 - may make sense to unify?

Reply:
Yep no worries.  Often now I'm starting to write integration tests by default - using unit tests only for testing internal APIs or more trivial things.

A big benefit of integration tests is that one is also testing whether all the APIs are public correctly, etc.

Reply:
(Perhaps it's also time to break up `main.rs` into separate test modules)